juju/model/__init__.py

```diff
@@ -3064,16 +3103,21 @@ async def wait_for_idle(
             wait_for_at_least_units if wait_for_at_least_units is not None else 1
         )
 
-        timeout = timedelta(seconds=timeout) if timeout is not None else None
-        idle_period = timedelta(seconds=idle_period)
+        timeout_ = timedelta(seconds=timeout) if timeout is not None else None
+        idle_period_ = timedelta(seconds=idle_period)
         start_time = datetime.now()
-        # Type check against the common error of passing a str for apps
-        if apps is not None and (
-            not isinstance(apps, list) or any(not isinstance(o, str) for o in apps)
-        ):
-            raise JujuError(f"Expected a List[str] for apps, given {apps}")
 
-        apps = apps or self.applications
+        if isinstance(apps, (str, bytes, bytearray, memoryview)):
+            raise ValueError("apps must be a Sequence[str] and not a plain type")
+
+        apps_ = apps or list(self.applications)
```

Comment:
I could look and try to figure this out but I'll just be lazy and ask -- why the change from `self.applications` to `list(self.applications)`?

Reply:
because of the type hint really, as self.applications is a dict, and keys are something special... anyway both are iterable, but type hints force some kind of cast.